#! /bin/sh -e
## 33_sb1_oprofile.dpatch by Mark Mason <mason@broadcom.com>
##
## DP: SB1: Add oprofile support.
## DP: Upstream status: in linux-mips git

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

--- 2.6.15/arch/mips/oprofile/Makefile	2006-01-10 11:21:16.000000000 +0000
+++ git/arch/mips/oprofile/Makefile	2006-01-19 22:17:59.000000000 +0000
@@ -12,4 +12,5 @@
 
 oprofile-$(CONFIG_CPU_MIPS32)		+= op_model_mipsxx.o
 oprofile-$(CONFIG_CPU_MIPS64)		+= op_model_mipsxx.o
+oprofile-$(CONFIG_CPU_SB1)		+= op_model_mipsxx.o
 oprofile-$(CONFIG_CPU_RM9000)		+= op_model_rm9000.o
--- 2.6.15/arch/mips/oprofile/common.c	2006-01-10 11:21:16.000000000 +0000
+++ git/arch/mips/oprofile/common.c	2006-01-19 22:17:59.000000000 +0000
@@ -79,6 +79,8 @@
 	case CPU_20KC:
 	case CPU_24K:
 	case CPU_25KF:
+	case CPU_SB1:
+	case CPU_SB1A:
 		lmodel = &op_model_mipsxx;
 		break;
 
--- 2.6.15/arch/mips/oprofile/op_model_mipsxx.c	2006-01-10 11:21:16.000000000 +0000
+++ git/arch/mips/oprofile/op_model_mipsxx.c	2006-01-19 22:17:59.000000000 +0000
@@ -205,6 +205,11 @@
 		op_model_mipsxx.cpu_type = "mips/5K";
 		break;
 
+	case CPU_SB1:
+	case CPU_SB1A:
+		op_model_mipsxx.cpu_type = "mips/sb1";
+		break;
+
 	default:
 		printk(KERN_ERR "Profiling unsupported for this CPU\n");
 

